### ISSUE TITLE
CalDav not working after upgrade from 8.2.3 to 9.0 or fresh install of 9.0 on Ubuntu

### ISSUE THREAD

**Issue by DeathByDenim** [owncloud/core#23745]
### Steps to reproduce
1. Install owncloud 9.0 on Kubuntu 15.10 using apache2 and mysql
2. Enable the calendar and contacts app
3. Add caldav calendar to KOrganizer (or akonadi)

### Expected behaviour
Calendars should show up in KOrganizer

### Actual behaviour
No external calendars are added. Retrieving the collection from the akonadi set up just returns an empty list as if no calendars exist even though it can connect to the server just fine.

I tried with cadaver too, which results in:
```
dav:/owncloud/remote.php/caldav/> ls
Listing collection `/owncloud/remote.php/caldav/': succeeded.
Coll:   calendars                              0  dec 31  1969
Coll:   principals                             0  dec 31  1969
dav:/owncloud/remote.php/caldav/> cd calendars
dav:/owncloud/remote.php/caldav/calendars/> ls
Listing collection `/owncloud/remote.php/caldav/calendars/': failed:                                                                                 
405 Method Not Allowed                                                                                                                               
```

This is on a fresh virtual machine with kubuntu 15.10 for testing purposes, but it happens on my production machine as well that got upgraded from owncloud 8.2.3 to 9.0, after which the calendars and contacts stopped syncing. Re-adding the caldav in korganizer just made the calendars disappear.

I hope this isn't a duplicate report, but I didn't see any with specifically these symptoms.
### Server configuration
**Operating system**:
Kubuntu 15.10
**Web server:**
Apache2 2.4.12-2ubuntu2
**Database:**
Mysql 5.6.28-0ubuntu0.15.10.1
**PHP version:**
5.6.11+dfsg-1ubuntu3.1
**ownCloud version:** (see ownCloud admin page)
9.0.0-3.1
**Updated from an older ownCloud or fresh install:**
Fresh install
**Where did you install ownCloud from:**
Added apt repositories from owncloud.org
**Signing status (ownCloud 9.0 and above):**
no errors found

**List of activated apps:**
  - activity: 2.2.1
  - calendar: 1.0
  - comments: 0.2
  - contacts: 1.1.0.0
  - dav: 0.1.5
  - federatedfilesharing: 0.1.0
  - federation: 0.0.4
  - files: 1.4.4
  - files_pdfviewer: 0.8
  - files_sharing: 0.9.1
  - files_texteditor: 2.1
  - files_trashbin: 0.8.0
  - files_versions: 1.2.0
  - files_videoplayer: 0.9.8
  - firstrunwizard: 1.1
  - gallery: 14.5.0
  - notifications: 0.2.3
  - provisioning_api: 0.4.1
  - systemtags: 0.2
  - templateeditor: 0.1
  - updatenotification: 0.1.0

**The content of config/config.php:**
```
{
    "system": {
        "updatechecker": false,
        "instanceid": "occtzselip49",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "kubuntu1510"
        ],
        "datadirectory": "\/var\/www\/owncloud\/data",
        "overwrite.cli.url": "https:\/\/kubuntu1510\/owncloud",
        "dbtype": "mysql",
        "version": "9.0.0.19",
        "dbname": "owncloud",
        "dbhost": "localhost",
        "dbtableprefix": "oc_",
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "logtimezone": "UTC",
        "installed": true
    }
}
```
**Are you using external storage, if yes which one:** local/smb/sftp/...
No
**Are you using encryption:** yes/no
No
**Are you using an external user-backend, if yes which one:** LDAP/ActiveDirectory/Webdav/...
No
### Client configuration
**Browser:**
KOrganizer
**Operating system:**
Kubuntu 15.10
#### Web server error log
```
127.0.0.1 - - [01/Apr/2016:15:30:48 -0400] "PROPFIND /owncloud/remote.php/carddav/ HTTP/1.1" 401 3694 "-" "Mozilla/5.0 (X11; Linux x86_64) KHTML/5.15.0 (like Gecko) Konqueror/5.15"
127.0.0.1 - - [01/Apr/2016:15:30:48 -0400] "PROPFIND /owncloud/remote.php/caldav/ HTTP/1.1" 401 3700 "-" "Mozilla/5.0 (X11; Linux x86_64) KHTML/5.15.0 (like Gecko) Konqueror/5.15"
127.0.0.1 - jarno [01/Apr/2016:15:30:48 -0400] "PROPFIND /owncloud/remote.php/carddav/ HTTP/1.1" 207 4145 "-" "Mozilla/5.0 (X11; Linux x86_64) KHTML/5.15.0 (like Gecko) Konqueror/5.15"
127.0.0.1 - jarno [01/Apr/2016:15:30:48 -0400] "PROPFIND /owncloud/remote.php/caldav/ HTTP/1.1" 207 4206 "-" "Mozilla/5.0 (X11; Linux x86_64) KHTML/5.15.0 (like Gecko) Konqueror/5.15"
127.0.0.1 - - [01/Apr/2016:15:30:48 -0400] "PROPFIND /owncloud/remote.php/carddav/principals/users/jarno/ HTTP/1.1" 401 3698 "-" "Mozilla/5.0 (X11; Linux x86_64) KHTML/5.15.0 (like Gecko) Konqueror/5.15"
127.0.0.1 - - [01/Apr/2016:15:30:48 -0400] "PROPFIND /owncloud/remote.php/caldav/principals/users/jarno/ HTTP/1.1" 401 3698 "-" "Mozilla/5.0 (X11; Linux x86_64) KHTML/5.15.0 (like Gecko) Konqueror/5.15"
127.0.0.1 - jarno [01/Apr/2016:15:30:48 -0400] "PROPFIND /owncloud/remote.php/carddav/principals/users/jarno/ HTTP/1.1" 404 3686 "-" "Mozilla/5.0 (X11; Linux x86_64) KHTML/5.15.0 (like Gecko) Konqueror/5.15"
127.0.0.1 - jarno [01/Apr/2016:15:30:48 -0400] "PROPFIND /owncloud/remote.php/caldav/principals/users/jarno/ HTTP/1.1" 404 3686 "-" "Mozilla/5.0 (X11; Linux x86_64) KHTML/5.15.0 (like Gecko) Konqueror/5.15"
127.0.0.1 - - [01/Apr/2016:15:30:48 -0400] "PROPFIND /owncloud/remote.php/carddav/ HTTP/1.1" 401 3706 "-" "Mozilla/5.0 (X11; Linux x86_64) KHTML/5.15.0 (like Gecko) Konqueror/5.15"
127.0.0.1 - - [01/Apr/2016:15:30:48 -0400] "PROPFIND /owncloud/remote.php/caldav/ HTTP/1.1" 401 3698 "-" "Mozilla/5.0 (X11; Linux x86_64) KHTML/5.15.0 (like Gecko) Konqueror/5.15"
127.0.0.1 - jarno [01/Apr/2016:15:30:48 -0400] "PROPFIND /owncloud/remote.php/carddav/ HTTP/1.1" 207 4793 "-" "Mozilla/5.0 (X11; Linux x86_64) KHTML/5.15.0 (like Gecko) Konqueror/5.15"
127.0.0.1 - jarno [01/Apr/2016:15:30:48 -0400] "PROPFIND /owncloud/remote.php/caldav/ HTTP/1.1" 207 5505 "-" "Mozilla/5.0 (X11; Linux x86_64) KHTML/5.15.0 (like Gecko) Konqueror/5.15"
127.0.0.1 - - [01/Apr/2016:15:30:51 -0400] "GET /owncloud/ocs/v2.php/apps/notifications/api/v1/notifications?format=json HTTP/1.1" 200 1040 "-" "Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:45.0) Gecko/20100101 Firefox/45.0"
127.0.0.1 - - [01/Apr/2016:15:31:21 -0400] "GET /owncloud/ocs/v2.php/apps/notifications/api/v1/notifications?format=json HTTP/1.1" 200 1040 "-" "Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:45.0) Gecko/20100101 Firefox/45.0"
127.0.0.1 - - [01/Apr/2016:15:31:51 -0400] "GET /owncloud/ocs/v2.php/apps/notifications/api/v1/notifications?format=json HTTP/1.1" 200 3117 "-" "Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:45.0) Gecko/20100101 Firefox/45.0"
```
#### ownCloud log (data/owncloud.log)
```
{"reqId":"p43y9VNlcymcojYJRgjA","remoteAddr":"127.0.0.1","app":"PHP","message":"Undefined offset: 0 at \/var\/www\/owncloud\/lib\/private\/setup\/mysql.php#46","level":3,"time":"2016-04-01T18:52:20+00:00"}
{"reqId":"p43y9VNlcymcojYJRgjA","remoteAddr":"127.0.0.1","app":"no app in context","message":"Exception: {\"Exception\":\"DomainException\",\"Message\":\"Contacts tables are missing. Nothing to do.\",\"Code\":0,\"Trace\":\"#0 \\\/var\\\/www\\\/owncloud\\\/apps\\\/dav\\\/lib\\\/migration\\\/migrateaddressbooks.php(83): OCA\\\\Dav\\\\Migration\\\\AddressBookAdapter->setup()\\n#1 \\\/var\\\/www\\\/owncloud\\\/apps\\\/dav\\\/appinfo\\\/application.php(186): OCA\\\\Dav\\\\Migration\\\\MigrateAddressbooks->setup()\\n#2 \\\/var\\\/www\\\/owncloud\\\/apps\\\/dav\\\/appinfo\\\/install.php(26): OCA\\\\Dav\\\\AppInfo\\\\Application->migrateAddressbooks()\\n#3 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/installer.php(621): include('\\\/var\\\/www\\\/ownclo...')\\n#4 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/installer.php(572): OC_Installer::includeAppScript('\\\/var\\\/www\\\/ownclo...')\\n#5 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/installer.php(546): OC_Installer::installShippedApp('dav')\\n#6 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/setup.php(370): OC_Installer::installShippedApps()\\n#7 \\\/var\\\/www\\\/owncloud\\\/core\\\/controller\\\/setupcontroller.php(64): OC\\\\Setup->install(Array)\\n#8 \\\/var\\\/www\\\/owncloud\\\/lib\\\/base.php(823): OC\\\\Core\\\\Controller\\\\SetupController->run(Array)\\n#9 \\\/var\\\/www\\\/owncloud\\\/index.php(39): OC::handleRequest()\\n#10 {main}\",\"File\":\"\\\/var\\\/www\\\/owncloud\\\/apps\\\/dav\\\/lib\\\/migration\\\/addressbookadapter.php\",\"Line\":72}","level":3,"time":"2016-04-01T18:52:29+00:00"}
{"reqId":"p43y9VNlcymcojYJRgjA","remoteAddr":"127.0.0.1","app":"no app in context","message":"Exception: {\"Exception\":\"DomainException\",\"Message\":\"Calendar tables are missing. Nothing to do.\",\"Code\":0,\"Trace\":\"#0 \\\/var\\\/www\\\/owncloud\\\/apps\\\/dav\\\/lib\\\/migration\\\/migratecalendars.php(84): OCA\\\\Dav\\\\Migration\\\\CalendarAdapter->setup()\\n#1 \\\/var\\\/www\\\/owncloud\\\/apps\\\/dav\\\/appinfo\\\/application.php(202): OCA\\\\Dav\\\\Migration\\\\MigrateCalendars->setup()\\n#2 \\\/var\\\/www\\\/owncloud\\\/apps\\\/dav\\\/appinfo\\\/install.php(27): OCA\\\\Dav\\\\AppInfo\\\\Application->migrateCalendars()\\n#3 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/installer.php(621): include('\\\/var\\\/www\\\/ownclo...')\\n#4 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/installer.php(572): OC_Installer::includeAppScript('\\\/var\\\/www\\\/ownclo...')\\n#5 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/installer.php(546): OC_Installer::installShippedApp('dav')\\n#6 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/setup.php(370): OC_Installer::installShippedApps()\\n#7 \\\/var\\\/www\\\/owncloud\\\/core\\\/controller\\\/setupcontroller.php(64): OC\\\\Setup->install(Array)\\n#8 \\\/var\\\/www\\\/owncloud\\\/lib\\\/base.php(823): OC\\\\Core\\\\Controller\\\\SetupController->run(Array)\\n#9 \\\/var\\\/www\\\/owncloud\\\/index.php(39): OC::handleRequest()\\n#10 {main}\",\"File\":\"\\\/var\\\/www\\\/owncloud\\\/apps\\\/dav\\\/lib\\\/migration\\\/calendaradapter.php\",\"Line\":68}","level":3,"time":"2016-04-01T18:52:29+00:00"}
```

**Comment by DeepDiver1975** [comment]
> Listing collection `/owncloud/remote.php/caldav/calendars/': failed: 405 Method Not Allowed

listing users is not allowed - just cd into the collection with the name of the user who is logged in

**Comment by DeepDiver1975** [comment]
duplicate ? #23715

**Comment by DeathByDenim** [comment]
Yeah, I saw that one. It's close, but I don't see the calendars at all. It may very well be related of course.
Sorry about the misinformation with cadaver. I assumed akonadi was doing the same thing. But you are right, if I cd into "username" and use ls there, it does list the available calendars. I'm also able to download calendar events from there. Could this be an issue with KOrganizer only? I've tried it both in Kubuntu 14.04 and 15.10 and even the 16.04 test-build.

I can provide a virtual machine that exhibits this behaviour if that's helpful. It's where I reproduced this issue.
